Library/ViewModels/FindFriendsViewModel.swift

```diff
@@ -139,7 +143,14 @@ public final class FindFriendsViewModel: FindFriendsViewModelType, FindFriendsVi
         .materialize()
     }
 
-    self.stats = Signal.combineLatest(statsEvent.values(), source)
+    let stats = Signal.combineLatest(statsEvent.values(), source)
+
+    self.stats = Signal.combineLatest(
+        shouldShowFacebookConnect,
+        stats
+      ).map(unpack) // unpack into 3-tuple (shouldShowFacebookConnect, friends, source)
+      .filter { !$0.0 } // filter for when shouldShowFacebookConnect is false
+      .map { ($1, $2)} // map back into expected tuple (friends, source)
```

Comment:
Here I would write the name of the values instead of using the shorthand syntax:
```
self.stats = Signal.combineLatest(
        shouldShowFacebookConnect,
        stats
      ).map(unpack) // unpack into 3-tuple (shouldShowFacebookConnect, friends, source)
      .filter { shouldShowFacebookConnect in  !shouldShowFacebookConnect }
      .map { friends, source in  
         (friends, source)
      } 
```
This way the code becomes self-explanatory and we won't need the comments anymore.